🐛 Bug: permissions: id-write missing from release.yml

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

Packages should be able to auto-publish. As of #461, that includes package provenance. 

### Actual

https://github.com/JoshuaKGoldberg/cached-factory/actions/runs/5988631169/job/16244072020:

```plaintext
npm ERR! Provenance generation in GitHub Actions requires "write" access to the "id-token" permission
```

### Additional Info

_No response_

--- a/src/steps/writing/creation/dotGitHub/createWorkflowFile.ts
+++ b/src/steps/writing/creation/dotGitHub/createWorkflowFile.ts
@@ -27,6 +27,7 @@ interface WorkflowFileOn {
 
 interface WorkflowFilePermissions {
 	contents?: string;
+	"id-token"?: string;
 	"pull-requests"?: string;
 }
 

--- a/src/steps/writing/creation/dotGitHub/workflows.ts
+++ b/src/steps/writing/creation/dotGitHub/workflows.ts
@@ -158,6 +158,7 @@ export function createWorkflows(options: Options) {
 				},
 				permissions: {
 					contents: "write",
+					"id-token": "write",
 				},
 				steps: [
 					{

